app/main.py

```diff
@@ -2,0 +1,96 @@
+from app.cafe import Cafe
+from app.errors import (NotVaccinatedError, OutdatedVaccineError,
+                        NotWearingMaskError)
+import datetime
+
+
+def go_to_cafe(friends: list[dict], cafe: Cafe) -> str:
+    masks_needed = 0
+    all_vaccinated = True
+
+    for friend in friends:
+        try:
+            if "vaccine" not in friend:
+                all_vaccinated = False
+                continue
+
+            expiration_date = friend["vaccine"]["expiration_date"]
+            if expiration_date < datetime.date.today():
+                all_vaccinated = False
```

Comment:
Instead of manually setting `all_vaccinated` to `False` when the vaccine is outdated, you should raise an `OutdatedVaccineError`. This will allow the `try/except` block to handle the error as intended.